### PR TITLE
publish to ftp

### PR DESCRIPTION
**Pull request by hugoqui**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Publish to FTP on Development Push
```
Comment (la-senda-antigua) [inline .github/workflows/main.yml]: change the name of the file to something more descriptive. What is this yaml file for?
maybe something like deploy_plan?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches:
+      - development
```
Comment (la-senda-antigua) [inline .github/workflows/main.yml]: we need to run auto deployment for production too. Do you need to create another yaml file for the main branch or do you want to modify this one to work for both development and main?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  publish:
+    runs-on: windows-latest
```
Comment (la-senda-antigua) [inline .github/workflows/main.yml]: i created two environments: test and production. You need to specify what environment this is for, so the workflow knows what variables/secrets to use.
https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Build and publish
+        run: |
+          msbuild /p:Configuration=Release /p:OutputPath=./publish_output
+          echo "::set-output name=publish_path::$(pwd)/publish_output"
+
+      - name: Upload to FTP
+        uses: SamKirkland/FTP-Deploy-Action@3.1.1
+        with:
+          server: ${{ secrets.FTP_SERVER }}
+          username: ${{ secrets.FTP_USERNAME }}
```
Comment (la-senda-antigua) [inline .github/workflows/main.yml]: I added only password as a secret. FTP_SERVER and FTP_USERNAME are under variables.

File: .github/workflows/main.yml (hunk 2 above)
Comment (la-senda-antigua) [reply .github/workflows/main.yml]: also, add the worflow_dispatch trigger so we can run this manually.
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+          password: ${{ secrets.FTP_PASSWORD }}
+          local-dir: ${{ steps.build.outputs.publish_path }}
+          remote-dir: '/test-services.iglesialasendaantigua.com'
```
Comment (la-senda-antigua) [inline .github/workflows/main.yml]: this should be in an environment variable.

**Review by hugoqui — COMMENTED** [review]
pr resolve

File: .github/workflows/deploy_plan.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Build and publish
+        run: |
+          msbuild /p:Configuration=Release /p:OutputPath=./publish_output
+          echo "::set-output name=publish_path::$(pwd)/publish_output"
+
+      - name: Upload to FTP
+        uses: SamKirkland/FTP-Deploy-Action@3.1.1
+        with:
+          server: iglesiacristianabelen.com
```
Comment (la-senda-antigua) [inline .github/workflows/deploy_plan.yml]: is this a mistake?

File: .github/workflows/main.yml (hunk 2 above)
Comment (hugoqui) [reply .github/workflows/main.yml]: Done

**Review by la-senda-antigua — APPROVED** [review]

**Pull request merged**